src/coreclr/vm/syncblk.inl

```diff
@@ -683,6 +687,7 @@ FORCEINLINE AwareLock::LeaveHelperAction AwareLock::LeaveHelper(Thread* pCurThre
         MODE_ANY;
     } CONTRACTL_END;
 
+    // if (m_HoldingThreadId != pCurThread->GetThreadId()) some tests failed when using this line instead
```

Comment:
Why did the test fail?

Reply:
Not sure yet, will look into that

Reply:
Actually there was some line missing in AwareLock::LeaveHelper which caused the failure